app/models/non_hmis_client.rb

```diff
@@ -73,6 +73,10 @@ def cohort_names
     arel_table[:client_identifier].lower.matches("%#{name.downcase}%")
   end
 
+  def self.data_source_id
```

Comment:
Perhaps this could be 
```
def self.deidentified_data_source
  DataSource.where(db_identifier: 'Deidentified')
end
```
Then above and below you could call `where(data_source_id: NonHmisClient.deidentified_data_source.select(:id))`
This will centralize the complexity, but keep a relation and prevent an extra query.